sql/core/src/main/java/org/apache/spark/sql/sources/v2/ContinuousReadSupport.java

```diff
@@ -0,0 +1,25 @@
+package org.apache.spark.sql.sources.v2;
+
+import java.util.Optional;
+
+import org.apache.spark.sql.sources.v2.reader.ContinuousReader;
+import org.apache.spark.sql.sources.v2.reader.DataSourceV2Reader;
+import org.apache.spark.sql.types.StructType;
+
+/**
+ * A mix-in interface for {@link DataSourceV2}. Data sources can implement this interface to
+ * provide data reading ability for continuous stream processing.
+ */
+public interface ContinuousReadSupport extends DataSourceV2 {
+  /**
+   * Creates a {@link DataSourceV2Reader} to scan the data from this data source.
+   *
+   * @param schema the user provided schema, or empty() if none was provided
+   * @param checkpointLocation a path to HDFS scratch space that can be used for failure recovery.
```

Comment:
`Hadoop FS scratch space`? It can be HDFS or S3 or other file systems that implements the Hadoop FS interface.